Sources/NIO/System.swift

```diff
@@ -113,43 +113,32 @@ private func preconditionIsNotBlacklistedErrno(err: CInt, where function: String
     precondition(!isBlacklistedErrno(err), "blacklisted errno \(err) \(String(cString: strerror(err)!)) in \(function))")
 }
 
-/* Sorry, we really try hard to not use underscored attributes. In this case however we seem to break the inlining threshold which makes a system call take twice the time, ie. we need this exception. */
-@inline(__always)
-internal func wrapSyscallMayBlock<T: FixedWidthInteger>(where function: String = #function, _ body: () throws -> T) throws -> IOResult<T> {
-    while true {
-        let res = try body()
-        if res == -1 {
-            let err = errno
-            switch err {
-            case EINTR:
-                continue
-            case EWOULDBLOCK:
-                return .wouldBlock(0)
-            default:
-                preconditionIsNotBlacklistedErrno(err: err, where: function)
-                throw IOError(errnoCode: err, reason: function)
-            }
-
-        }
-        return .processed(res)
-    }
+internal extension IOResult where T: FixedWidthInteger {
+  var result: T! {
+    get {
+      switch self {
+      case .processed(let result):
+        return result
+      default:
+        return nil
+      }
+    }
+  }
 }
 
 /* Sorry, we really try hard to not use underscored attributes. In this case however we seem to break the inlining threshold which makes a system call take twice the time, ie. we need this exception. */
 @inline(__always)
-@discardableResult
-internal func wrapSyscall<T: FixedWidthInteger>(where function: String = #function, _ body: () throws -> T) throws -> T {
+internal func call<T: FixedWidthInteger>(nonblocking: Bool, where function: String = #function, _ body: () throws -> T) throws -> IOResult<T> {
```

Comment:
I think we should still name this `wrapSyscall` or `callSyscall` or something. This wrapper only works if the called function behaves like a unix syscall (ie. -1 == error)